src/lib/Poster.svelte

```diff
@@ -38,6 +40,8 @@
   const title = media.title || media.name;
   const poster = `https://image.tmdb.org/t/p/w500${media.poster_path}`;
   const link = media.id ? `/${media.media_type}/${media.id}` : undefined;
+  const dateStr = media.release_date || media.first_air_date;
+  const year = dateStr ? new Date(dateStr).getFullYear() : undefined;
```

Comment:
```suggestion
  const year = new Date(media.release_date || media.first_air_date)?.getFullYear();
```

Not an important change, but I think it can help simplify (of course if you make this change, need to remove the `dateStr` variable too).

I removed the use of the ternary operator (for year), It is probably not worth setting it to undefined, if the method doesn't work, it will be already.

Reply:
My understanding is that passing in an undefined value, as we would here `new Date(undefined || undefined)` would result in an invalid date, which TS won't allow. I'd propose `const year = (media. release_date || media. first_air_date) && new Date ((media. release_date || media. first_air_date) !) .getFullYear) ;` if you want to eliminate the dateStr variable. Thoughts?

Reply:
@kurtmgray Ah I see, we should probably just keep it how it was then in that case.

Reply:
Great, not a problem. I went with `const year = dateStr && new Date(dateStr).getFullYear();` which will give us the `string | undefined` type we want, dependent upon `dateStr`.